ec/ecresource/deploymentresource/apm/v2/schema.go

```diff
@@ -112,6 +112,13 @@ func ApmSchema() schema.Attribute {
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"instance_configuration_version": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
```

Comment:
So I think this (and the same in the other non-elasticsearch topology elements) may blow up when the `deployment.deployment_template_id` attribute is changed, since the config version may change as a result. 

The elasticsearch schema uses `UseTopologyStateForUnknown` instead which explicitly handles this situation (setting the `instance_configuration_version` to unknown if the top level template is changed). I think we'll want to do something similar here.

Reply:
This a great point, thanks for bringing it up! I made the change in https://github.com/elastic/terraform-provider-ec/pull/755/commits/2cc4ec49892415293c56286ae3f44f2ee616524c.